### DISCUSSION TITLE
Excluding a File from Renaming in Template Generation 😭

### DISCUSSION THREAD

**Discussion by sorokin-vs** [dotnet/templating#7872] [General]
Hi, I've encountered a challenge that I could use some help with.

I've created this template, which can be found here:
https://github.com/bohems/service-template

Subsequently, I generate a template using the following command:
`dotnet new service -n ServiceName -F net8.0`

This results in all instances of '`Template`' being replaced with '`ServiceName`' which is exactly what I want.

However, within this repository, there's a specific file I'm concerned about:
https://github.com/bohems/service-template/blob/main/Template/Tools/Template.DbSeeder/seed-settings.json

Within this file, there's a key named "`outputTemplate`", and unfortunately, it gets renamed too, which is not what I intend:
```
{
  "Serilog": {
    "MinimumLevel": "Information",
    "WriteTo": [
      {
        "Name": "Console",
        "Args": {
          "formatter": "",
          // In this key, the word "Template" gets renamed, but I do not want this to occur.
          "outputTemplate": "[{Timestamp:HH:mm:ss} {Level:u3}] {Message:lj}{NewLine}{Exception}"
        }
      }
    ]
  },
  "ConnectionStrings": {
    "DefaultConnection": "",
  }
}
```

Could you provide guidance on how to address this issue, perhaps a way to exclude this file from being renamed, or a workaround to prevent renaming of that particular key?

**Comment by sayedihashimi** [comment]
There are ways that you can instruct the Template Engine to not perform the replacement on `outputTemplate`. For example, there are some methods listed at https://github.com/dotnet/templating/wiki/Conditional-processing-and-comment-syntax.

But I would say that having a sourceName of `Template` is not ideal. You should try to make that a unique value, and to include some special characters, so that you can avoid these kinds of issues. I have written up about `sourceName` at https://github.com/sayedihashimi/template-sample?tab=readme-ov-file#sourcename. 

If you can change the sourceName that would be the best approach, but if there is some constraint preventing that, take a look at using 
`copyOnly` or `cnd:noEmit` (more info in the first link that I shared).
